### PR TITLE
refactor(Badge): updated Badge variants' color names 

### PR DESCRIPTION
**Pull request by LinKCoding**
### Overview
Addressing [WEB-2446](https://codecademy.atlassian.net/browse/WEB-2446) -- 
Adjust dark-mode`grey` badges variants to meet WCAG contrast. 

Expanded the scope to update all variants names, see figma attached: 

### PR Checklist

- [x] Related to designs: https://www.figma.com/file/ReGfRNillGABAj5SlITalN/%F0%9F%93%90-Gamut?type=design&node-id=29959-39721&mode=design&t=InPLnttaWm84srkv-0 
- [x] Related to JIRA ticket: [WEB-2446]
- [x] I have run this code to verify it works
- [x] This PR includes unit tests for the code change
- [x] This PR includes testing instructions tests for the code change
- [x] The alpha package of this PR is passing end-to-end tests in all relevant Codecademy repositories

#### Testing Instructions

Load up storybook and find the badges page: http://localhost:6006/?path=/docs/atoms-badge--badge 
The badges should still load just fine. 

#### PR Links and Envs

| Repository   | PR Link                                                  | PR Env                                                   |
| :----------- | :------------------------------------------------------- | :------------------------------------------------------- |
| Monolith     | [Monolith PR](https://github.com/codecademy-engineering/Codecademy/pull/36634)  | [Monolith Env](https://pr-36634-monolith.dev-eks.codecademy.com/catalog?PR_ENV=pr-36634-portal) |
| Portal       | [Portal Link](https://github.com/codecademy-engineering/mono/pull/3934)  | [Portal Env](https://tengu.codecademy.com/catalog?PR_ENV=portal-app-pr-3934)   |

[WEB-2446]: https://codecademy.atlassian.net/browse/WEB-2446?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[WEB-2446]: https://codecademy.atlassian.net/browse/WEB-2446?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut@55.7.4-alpha.6f64f8.0<br/>@codecademy/gamut-kit@0.6.353-alpha.6f64f8.0<br/>@codecademy/styleguide@66.5.5-alpha.6f64f8.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6524cb266b6e967a058d4846--gamut-preview.netlify.app>https://6524cb266b6e967a058d4846--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6524cb266b6e967a058d4846>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut@55.7.4-alpha.88333a.0<br/>@codecademy/gamut-kit@0.6.353-alpha.88333a.0<br/>@codecademy/styleguide@66.5.5-alpha.88333a.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6524d007cade847e7ed74e81--gamut-preview.netlify.app>https://6524d007cade847e7ed74e81--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6524d007cade847e7ed74e81>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut@55.8.2-alpha.21a8e1.0<br/>@codecademy/gamut-kit@0.6.357-alpha.21a8e1.0<br/>@codecademy/styleguide@66.5.9-alpha.21a8e1.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://652ca4da1393ec63e56b2bc3--gamut-preview.netlify.app>https://652ca4da1393ec63e56b2bc3--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/652ca4da1393ec63e56b2bc3>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut@55.8.2-alpha.a715f9.0<br/>@codecademy/gamut-kit@0.6.357-alpha.a715f9.0<br/>@codecademy/styleguide@66.5.9-alpha.a715f9.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://65309abcdf54425567d66872--gamut-preview.netlify.app>https://65309abcdf54425567d66872--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/65309abcdf54425567d66872>Deploy Logs</a></p>

File: packages/styleguide/stories/Atoms/Badge.stories.mdx [hunk 1]
```diff
@@ -30,24 +35,21 @@ export const BadgeTemplate = (args) => (
...
 ## Variants
 
-We support the following variants of Badges: `default`, `grey`, `stroke`, `strokeContrast`, and `yellow`.
+We support the following variants of Badges: `primary`, `secondary`, `tertiary`, and `accent`.
 
-- **Greyscale Badges** provide different levels of priority for the `default` (primary), `grey` (secondary), and `stroke` (tertiary) variants
+- **Greyscale Badges** provide different levels of priority for the `primary`, `secondary`, and `tertiary` variants
 - **Accent colors** like `yellow` are meant to be used sparingly so they can stand out against other elements.
 
 Additional variants should designed with both <LinkTo kind='Foundations/ColorMode'>ColorModes</LinkTo> in mind. For example, the `yellow` and `grey` variants work against both light and dark backgrounds, while the `default` and `stroke` variants both adapt to their ColorMode (which you can check out by using the toolbar at the top of this page!)
```
Comment (dreamwasp) [inline packages/styleguide/stories/Atoms/Badge.stories.mdx:43]: let's update the variant names in this paragraph as well

File: packages/styleguide/stories/Atoms/Badge.stories.mdx [hunk 2]
```diff
@@ -30,24 +35,21 @@ export const BadgeTemplate = (args) => (
...
     {BadgeTemplate.bind({})}
   </Story>
-  <Story name="stroke" args={{ variant: 'stroke' }}>
-    {BadgeTemplate.bind({})}
-  </Story>
-  <Story name="strokeContrast" args={{ variant: 'strokeContrast' }}>
+  <Story name="tertiary" args={{ variant: 'tertiary' }}>
     {BadgeTemplate.bind({})}
   </Story>
   <Story name="yellow" args={{ variant: 'yellow' }}>
```
Comment (dreamwasp) [inline packages/styleguide/stories/Atoms/Badge.stories.mdx:55]: this variant arg should be updated to `accent` and below (line 86) the `stroke` variant should be updated to `tertiary`

**Review by dreamwasp — CHANGES_REQUESTED** [review]
just have a couple tweaks to the storybook story then should be ready to ship!

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut@55.8.2-alpha.47787b.0<br/>@codecademy/gamut-kit@0.6.357-alpha.47787b.0<br/>@codecademy/styleguide@66.5.9-alpha.47787b.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6531435a8d203727df4b482e--gamut-preview.netlify.app>https://6531435a8d203727df4b482e--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6531435a8d203727df4b482e>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut@55.8.2-alpha.703bf0.0<br/>@codecademy/gamut-kit@0.6.357-alpha.703bf0.0<br/>@codecademy/styleguide@66.5.9-alpha.703bf0.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6537da6f26768805d4b9f768--gamut-preview.netlify.app>https://6537da6f26768805d4b9f768--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6537da6f26768805d4b9f768>Deploy Logs</a></p>

**Pull request merged**